src/runtime.ts

```diff
@@ -631,6 +586,143 @@ export async function createDefaultRuntime(
   };
 }
 
+/**
+ * create new runtime instance
+ *
+ * @param      {any}               web3           connected web3 instance
+ * @param      {DfsInterface}      dfs            interface for retrieving file from dfs
+ * @param      {any}               runtimeConfig  configuration values
+ * @return     {Promise<Runtime>}  runtime instance
+ */
+export async function createDefaultRuntime(
```

Comment:
300+ LOCs 🙈
would be really easier to read if some functionality could be outsourced in extra functions.

Reply:
Was already a huge function before i implemented my changes. Would be a bit of too much refactoring, because the function needs all the local parameters to stack all the instances together.